app/bot/broadcast_status.go

```diff
@@ -0,0 +1,115 @@
+package bot
+
+import (
+	"context"
+	"net/http"
+	"time"
+
+	log "github.com/go-pkgz/lgr"
+)
+
+const (
+	MsgBroadcastStarted  = "Вещание началось"
+	MsgBroadcastFinished = "Вещание завершилось"
+)
+
+// BroadcastStatus bot returns current broadcast status
+type BroadcastStatus struct {
+	// Params
+	broadcastUrl string        // Url for "ping"
+	pingInterval time.Duration // Ping interval
+	delayToOff   time.Duration // After this interval of not OK, status will be switcher to OFF
+
+	// State
+	status         bool  // current broadcast status
+	lastSentStatus *bool // by default this value is null to react on first message
+	offPeriod      time.Duration
+	lastCheck      time.Time
+
+	// ping func (for tests)
+	ping func(ctx context.Context, url string) bool
```

Comment:
I don't really see the need for this artificial thing. I'd just use httptest.Server and it will do the job just fine without faking ping method

Reply:
Fixed